src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -685,6 +697,15 @@ export default class DefaultAudioVideoController implements AudioVideoController
             }`
           );
         } else {
+          if (this.sessionStateController.state() === SessionStateControllerState.NotConnected) {
+            this.handleEvent('ConnectionFailed', {
+              meetingSessionStatusCode: `${MeetingSessionStatusCode[status.statusCode()]}`,
+            });
+          } else {
+            this.handleEvent('MeetingEndedWithError', {
```

Comment:
Finding a triggering place is tricky because `DefaultAudioVideoController` has multiple places stopping the session. Instead, how about implementing an `audioVideoDidStop` observer itself within the controller?

```
audioVideoDidStop(status: MeetingSessionStatus): void {
  if (status.isFailure() || status.isAudioFailure()) {
    this.handleEvent('MeetingEndedWithError', {
      meetingSessionStatusCode: `${MeetingSessionStatusCode[status.statusCode()]}`,
    });
  } else {
    this.handleEvent('MeetingEnded');
  }
}
```

Reply:
^ Or we can move the `audioVideoDidStop` call into a common function

```
private notifyStop(status: MeetingSessionStatusCode) {
  if (status.isFailure() || status.isAudioFailure()) {
    this.handleEvent('MeetingEndedWithError', {
      meetingSessionStatusCode: `${MeetingSessionStatusCode[status.statusCode()]}`,
    });
  } else {
    this.handleEvent('MeetingEnded');
  }

  this.forEachObserver(observer => {
    Maybe.of(observer.audioVideoDidStop).map(f => f.bind(observer)(status));
  });
}
```

Reply:
I moved audioVideoDidStop into the common function, so it should only call MeetingEndedWithError if notifyStop is called with a isFailure or isAudioConnectionFailure status code.